lnwallet/chanfunding/psbt_assembler.go

```diff
@@ -543,3 +544,55 @@ func verifyInputsSigned(ins []*wire.TxIn) error {
 	}
 	return nil
 }
+
+// verifyAllInputsSegWit makes sure all inputs to a transaction are SegWit
+// spends. This is a bit tricky because the PSBT spec doesn't require the
+// WitnessUtxo field to be set. Therefore if only a NonWitnessUtxo is given, we
+// need to look at it and make sure it's either a witness pkScript or a nested
+// SegWit spend.
+func verifyAllInputsSegWit(txIns []*wire.TxIn, ins []psbt.PInput) error {
+	for idx, in := range ins {
+		switch {
+		// The optimal case is that the witness UTXO is set explicitly.
+		case in.WitnessUtxo != nil:
+
+		// Only the non witness UTXO field is set, we need to inspect it
+		// to make sure it's not P2PKH or bare P2SH.
+		case in.NonWitnessUtxo != nil:
+			utxo := in.NonWitnessUtxo
+			txIn := txIns[idx]
+			txOut := utxo.TxOut[txIn.PreviousOutPoint.Index]
```

Comment:
This won't panic which is good, but `psbt.SumUtxoInputValues` has the same indexing and will panic on malformed PSBTs.  Could be fixed in a followup and version-bumped?  It's irrelevant to this PR but I found it while reviewing this one.  I could also patch it instead.

Reply:
A follow up for the next rc sounds good IMO.